File: Shift.Web/ClientApp/src/app/infrastracture/entities/auth/EmployeeRegisterVM.ts
import { UserRegisterVM } from "./UserRegisterVM";
import { AcademicDegree } from "../users/AcademicDegree";
import { AcademicRank } from "../users/AcademicRank";
import { JobPosition } from "../users/JobPosition";
import { Department } from "../university/Department";

export class EmployeeRegisterVM {
    public User: UserRegisterVM;

    public AcademicDegree: AcademicDegree;
    public AcademicRank: AcademicRank;
    public JobPosition: JobPosition;
    public Department: Department;
}

File: Shift.Infrastructure/Models/ViewModels/Auth/EmployeeRegisterVM.cs
using Shift.Infrastructure.Models.ViewModels.Data;

namespace Shift.Infrastructure.Models.ViewModels.Auth
{
	public class EmployeeRegisterVM
	{
		public UserRegisterVM User { get; set; }

		public AcademicDegreeVM AcademicDegree { get; set; }
		public AcademicRankVM AcademicRank { get; set; }
		public JobPositionVM JobPosition { get; set; }
		public DepartmentVM Department { get; set; }
	}
}
